sdks/java/io/iceberg/build.gradle

```diff
@@ -45,6 +45,7 @@ dependencies {
     implementation library.java.vendored_guava_32_1_2_jre
     implementation project(path: ":sdks:java:core", configuration: "shadow")
     implementation project(path: ":model:pipeline", configuration: "shadow")
+    implementation library.java.avro
```

Comment:
I may have missed it, but did we need this dependency for this PR?

Reply:
yeah, it's not needed indeed. Staged incorrectly.

Reply:
ahah! it seems that it's actually needed?

```
Execution failed for task ':sdks:java:io:iceberg:analyzeClassesDependencies'.
> Dependency analysis found issues.
  usedUndeclaredArtifacts
   - org.apache.avro:avro:1.11.3@jar
```
when I double checked locally I wasn't getting this...

Reply:
hmm that's interesting, I checked out the PR and ran `./gradlew :sdks:java:io:iceberg:analyzeClassesDependencies` and got the same error. I don't see any lines in this change that introduce Avro library.

Anyways, I think there's no harm in including it. will merge this